x-pack/plugins/data_enhanced/public/search/sessions_mgmt/lib/api.ts

```diff
@@ -0,0 +1,169 @@
+    try {
+      const currentDate = moment();
+      const expiresDate = moment(created);
+      const duration = moment.duration(expiresDate.diff(currentDate));
+
+      if (duration.asDays() <= EXPIRES_SOON_IN_DAYS) {
+        // TODO: handle negatives by setting status to expired?
+        expiresSoon = true;
+      }
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.error(`Could not calculate duration to expiration`);
+      // eslint-disable-next-line no-console
+      console.error(err);
+    }
+  }
+
+  // derive the URL and add it in
+  let url = '/';
+  try {
+    url = await urls.getUrlGenerator(urlGeneratorId).createUrl(restoreState);
+  } catch (err) {
+    // eslint-disable-next-line no-console
+    console.error('Could not create URL from restoreState');
+    // eslint-disable-next-line no-console
+    console.error(err);
+  }
+
+  return {
+    id: savedObject.id,
+    isViewable: true, // always viewable
+    name,
+    appId,
+    created,
+    expires,
+    status,
+    actions,
+    expiresSoon,
+    url,
+  };
+};
+
+export class SearchSessionsMgmtAPI {
+  constructor(
+    private sessionsClient: ISessionsClient,
+    private urls: UrlGeneratorsStart,
+    private notifications: NotificationsStart
+  ) {}
+
+  public async fetchTableData(): Promise<UISession[] | null> {
+    try {
+      const result = await this.sessionsClient.find({
+        page: 1,
+        perPage: SESSIONS_LISTING_SEARCH_SIZE,
+        sortField: 'created',
+        sortOrder: 'asc',
+      });
+      if (result.saved_objects) {
+        const savedObjects = result.saved_objects as BackgroundSessionSavedObject[];
+        return await Promise.all(savedObjects.map(mapToUISession(this.urls)));
+      }
+      return null;
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.error(err);
+
+      this.notifications.toasts.addError(err, {
+        title: i18n.translate('xpack.data.mgmt.searchSessions.api.fetchError', {
+          defaultMessage: 'Failed to refresh the page!',
+        }),
+      });
+
+      return null;
+    }
+  }
+
+  // Delete
+  public async sendDelete(id: string): Promise<UISession[] | null> {
+    try {
+      await this.sessionsClient.delete(id);
+
+      this.notifications.toasts.addSuccess({
+        title: i18n.translate('xpack.data.mgmt.searchSessions.api.deleted', {
+          defaultMessage: 'Deleted session',
+        }),
+      });
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.error(err);
+
+      this.notifications.toasts.addError(err, {
+        title: i18n.translate('xpack.data.mgmt.searchSessions.api.deleteError', {
+          defaultMessage: 'Failed to delete the session!',
+        }),
+      });
+    }
+
+    return await this.fetchTableData();
+  }
+
+  // Cancel: not implemented
+  public async sendCancel(id: string): Promise<UISession[] | null> {
+    this.notifications.toasts.addError(new Error('Not implemented'), {
```

Comment:
Totally not necessary in this PR, but I imagine we will eventually have `cancel` and `extend` on `sessionsClient` that just calls the associated route (similar to https://github.com/elastic/kibana/blob/eee4641cc177718f5aa15758057580f897cf6eac/src/plugins/data/public/search/session/sessions_client.ts#L82). Right now the route will return a 404 since it doesn't exist, but it might be useful to add.